lib/debugger/widgets/worldInspect.lua

```diff
@@ -9,68 +9,88 @@ return function(plasma)
 		local world = debugger.debugWorld
 
 		local cache, setCache = plasma.useState()
+		local ascendingOrder, _ = plasma.useState(false)
+		local skipIntersections, setSkipIntersections = plasma.useState(true)
 		local debugComponent, setDebugComponent = plasma.useState()
 
 		local closed = plasma
 			.window({
 				title = "World inspect",
 				closable = true,
 			}, function()
-				local skipIntersections
-
-				plasma.row(function()
-					plasma.heading("Size")
-					plasma.label(world:size())
-
-					plasma.space(30)
-					skipIntersections = plasma.checkbox("Hide intersecting components"):checked()
-
-					if plasma.button("view raw"):clicked() then
-						table.clear(objectStack)
-						objectStack[1] = {
-							value = world,
-							key = "Raw World",
-						}
-					end
-				end)
-
 				if not cache or os.clock() - cache.createdTime > debugger.componentRefreshFrequency then
 					cache = {
 						createdTime = os.clock(),
 						uniqueComponents = {},
+						emptyEntities = 0,
 					}
 
 					setCache(cache)
 
 					for _, entityData in world do
-						for component in entityData do
-							cache.uniqueComponents[component] = (cache.uniqueComponents[component] or 0) + 1
+						if next(entityData) == nil then
+							cache.emptyEntities += 1
+						else
+							for component in entityData do
+								cache.uniqueComponents[component] = (cache.uniqueComponents[component] or 0) + 1
+							end
 						end
 					end
 				end
 
+				plasma.row(function()
+					plasma.heading("Size")
+					plasma.label(
+						`{world:size()} {if cache.emptyEntities > 0 then `({cache.emptyEntities} empty)` else ""}`
+					)
+				end)
+
+				plasma.row({ padding = 15 }, function()
+					if plasma.checkbox("Show intersections", { checked = not skipIntersections }):clicked() then
+						setSkipIntersections(not skipIntersections)
+					end
+
+					if plasma.button("View Raw"):clicked() then
+						table.clear(objectStack)
+						objectStack[1] = {
+							value = world,
+							key = "Raw World",
+						}
+					end
+				end)
+
 				local items = {}
 				for component, count in cache.uniqueComponents do
 					table.insert(items, {
-						icon = count,
-						text = tostring(component),
-						component = component,
+						count,
+						tostring(component),
 						selected = debugComponent == component,
+						component = component,
```

Comment:
OOC, why is this a mixed table now?

Reply:
Plasma uses the numeric indices as the values it should show in the table and ignores the rest. `selected` is a special prop it uses to determine if this is the entry it should highlight, while `component` is custom data I attached that lets the other widgets know what component to inspect later on (otherwise we'd have to loop through all of the possible components to find it from the name.)

Reply:
It's a little bit confusing because this part used to be plugged into a different widget, `selectionList`, which had a different method of showing the items.